.pre-commit-config.yaml

```diff
@@ -0,0 +1,134 @@
+    files: \.h$
+
+  - id: clang-format
+    name: C-format-source
+    language: system
+    entry: scripts/pcf_clang_format.py
+    args: ["--style-file", "scripts/clang-format-c"]
+    files: \.c$
+    exclude: xnvme_fioe.c$
+
+  - id: clang-format
+    name: C-format-fioe
+    language: system
+    entry: scripts/pcf_clang_format.py
+    args: ["--style-file", "scripts/clang-format-fioe"]
+    files: xnvme_fioe\.c$
+
+# Git: intended for commit-message formating
+- repo: local
+  hooks:
+  - id: commit-msg-check
+    name: Git-Validate commit message
+    language: pygrep
+    entry: '\A(?![a-z]+:[a-z]+: .+\nSigned-off-by:.+)'
+    args: [--multiline]
+    stages: [commit-msg]
+
+# Yaml: intended for the GitHUB Actions .yaml
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.1.0
+  hooks:
+  - id: check-yaml # pre-commit-hook-yamlfmt only works on valid YAML files
+    name: YAML-format-check
+- repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt
+  rev: 0.1.0
+  hooks:
+  - id: yamlfmt
+    name: YAML-format
+    args:
+    - '--mapping=2'
+    - '--sequence=2'
+    - '--offset=0'
+    - '--width=120'
+    - '--preserve-quotes'
+    types: [yaml]
+
+# Python: for all the helpers in scripts/*
+- repo: https://github.com/psf/black
+  rev: '22.3.0'
+  hooks:
+  - id: black
+    name: Python-format-black
+- repo: https://github.com/pycqa/isort
+  rev: '5.10.1'
+  hooks:
+  # Available args: https://pycqa.github.io/isort/docs/configuration/options.html
+  - id: isort
+    name: Python-format-isort
+    args:
+    - '--profile=black'
+    - '--line-length=88'
+- repo: https://github.com/pycqa/flake8
+  rev: '4.0.1'
+  hooks:
+  - id: flake8
+    name: Python-lint-flake8
+    args:
+    - '--max-line-length=88'
+    - '--extend-ignore=E203' # ignore "whitespace before ':'"
+- repo: https://github.com/jendrikseipp/vulture
+  rev: 'v2.3'
+  hooks:
+  - id: vulture
+    name: Python-lint-vulture
+    types: [python]
+    pass_filenames: true
+    # excluding sphinx-conf as it would break to change it
+    exclude: docs/autogen/conf.py$
+
+# Because pre-commit runs mypy from an isolated virtualenv (without your dependencies) you may also
+# find it useful to add the typed dependencies to additional_dependencies so mypy can better
+# perform dynamic analysis
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: 'v0.942'
+  hooks:
+  - id: mypy
+    name: Python-lint-mypy
+    args:
+    - '--no-strict-optional'
+    - '--ignore-missing-imports'
+# pre-commit runs pylint from an isolated virtualenv. Many of pylint's checks perform dynamic
+# analysis which will fail there. You may find configuring pylint as a local hook more useful.
+- repo: https://github.com/pre-commit/mirrors-pylint
+  rev: 'v3.0.0a4'
+  hooks:
+  - id: pylint
+    name: Python-lint-pylint
+    args:
+    - '--jobs=4' # Run with 4 threads
+    - '--good-names=i,j,k,n,f,_,Run,ex' # Add 'f' for with open()
```

Comment:
`v` is also common in `for k,v in some_dict.items()`

Reply:
Fixing...

Reply:
Fixed.